### ISSUE TITLE
[TF2] Players using controllers are unable to vote

### ISSUE THREAD

**Issue by rowedahelicon** [ValveSoftware/Source-1-Games#4251]
Tested with the Steam Deck (SteamOS) and a PS5 controller (on Windows 10), the Yes/No vote binds (Defaulting to Left Pad / D-Pad) do not work because they use the old vote system.

The [June 21, 2022 update](https://wiki.teamfortress.com/wiki/June_21,_2022_Patch) introduced the possibility of having multiple votes running at once. The vote client command was changed to handle this by including an ID number with the command. 

Previously `vote option1`, Current `vote <vote id> option1`. Controller inputs that have default yes/no voting options bound are not passing the vote ID, so they cannot vote at all.

**Comment by LastTalon** [comment]
Per https://github.com/ValveSoftware/Source-1-Games/issues/4388#issuecomment-1273554569 this is the place for this discussion. So I'll re-iterate the additional parts of the issue here.

The issue seems to be wider in scope than not being able to vote with controller. As far as I can tell the current vote command can't be bound effectively for two reasons.

The new command requires a vote index that as far as I can tell is transient. In the command's current form (unless there is something I'm unaware of with the command, since its mostly undocumented) this can't be bound without auxiliary commands to update the index that should be used for the correct active vote.

The vote index (as far as I can tell, because its undocumented) can't be retrieved in any way using any command. So even if a decent scheme of binding were to be devised, the index would still not be able to be provided to it to bind correctly.

So as far as I can tell, unless there's some undocumented feature about the new command, this makes it completely unbindable.

**Comment by Thespikedballofdoom** [comment]
I've never had to deal with this because for some reason hud_reloadscheme enables my ability to vote when it doesn't for others. 
That being said, are these map index numbers by chance these ones?

[https://www.teamfortress.tv/61185/decoding-map-index-values-in-personal-game-data](https://www.teamfortress.tv/61185/decoding-map-index-values-in-personal-game-data)
![image](https://user-images.githubusercontent.com/43627353/196592802-a5228fce-a502-4677-8948-e9d875373405.png)

**Comment by LastTalon** [comment]
Any news on this? Doesn't seem to have changed or been addressed.

**Comment by JoshuaVandaele** [comment]
This issue is still up to date, could the title be modified to better reflect the issue at hand? Players currently cannot vote if they do not have a keyboard with a "F1" and "F2" key, and it cannot be rebound or called in console, which is a big issue if you have a keyboard without function keys or simply if one is broken, or, as the posts originally suggested, you are not even playing on a keyboard

**Comment by LastTalon** [comment]
Agreed. That's the issue I originally made, which was closed as a duplicate, so it seems like this issue should be modified to accurately reflect the issue.

**Comment by rowedahelicon** [comment]
@JoshuaVandaele @LastTalon All set!
